src/Microsoft.AspNetCore.Server.Kestrel/Networking/Libuv.cs

```diff
@@ -610,6 +674,9 @@ private static class NativeDarwinMonoMethods
             public static extern void uv_unref(UvHandle handle);
 
             [DllImport("__Internal", CallingConvention = CallingConvention.Cdecl)]
+            public static extern int uv_fileno(UvHandle handle, ref IntPtr socket);
```

Comment:
I think you are right. It's better and easier to have `uv_fileno` in __Internal just as any other function than do a special dance just to avoid it.